kwanj-k/ride-my-way-API-V1
# #158502253-Adds request related end-points

### What does this PR do?
Adds ride requests related endpoints

### Description of Task to be completed
* create end-point to create a ride request
* create end-point to get all ride requests
* create end-point to delete/cancel a ride request

### How should I test it?
 1. Clone the repo
  ```
  $ git clone https://github.com/kwanj-k/ride-my-way-API-V1.git
  $ cd into the cloned repo
  $ git checkout ft-requets-#158502253
  ```
Create a virtualenv and activate it::

    Virtualenv -p python3 .
    source bin/activate

Install the dependencies::

    pip install -r requirements.txt 

### Running tests without coverage
Run the tests from the terminal
```
python manage.py test
```

### Running tests with coverage
Run tests with coverage by running this command in the terminal
```
pytest --cov-report term-missing --cov=app
```
## Testing on Postman
Run the development server:
  ```
  $ flask run
  ```
Start Postman

* Make a `POST` request to `/api/v1/rides/1/requests`
* Make a `GET` request to `/api/v1/requests`
* Make a `DELETE` request to `/api/v1/requests/id`

All the end-points should respond accordingly


### What PT stories?
[#158502253]https://www.pivotaltracker.com/story/show/158502253

[#158557140]https://www.pivotaltracker.com/story/show/158557140

[#158576365]https://www.pivotaltracker.com/story/show/158576365
